File: summary/ts/17_shakutori.ts
function shakutori(arr: number[], k: number): number {
  const n = arr.length;
  let res = 0;
  let right = 0;
  let sum = 0;

  for (let left = 0; left < n; left++) {
    while (right < n && sum + arr[right] <= k) {
      sum += arr[right];
      right++;
    }

    res += right - left;

    if (right == left) {
      right++;
    } else {
      sum -= arr[left];
    }
  }

  return res;
}

// Example usage:
const inputArray = [1, 2, 3, 4, 5]; // Example input array
const k = 5; // Example k value
const result = shakutori(inputArray, k);
console.log(`Number of subarrays with sum <= ${k}: ${result}`);

File: summary/python/17_shakutori.py
def shakutori(arr, k):
    n = len(arr)
    res = 0
    right = 0
    sum = 0

    for left in range(n):
        while right < n and sum + arr[right] <= k:
            sum += arr[right]
            right += 1

        res += right - left

        if right == left:
            right += 1
        else:
            sum -= arr[left]

    return res

# Example usage:
input_array = list(map(int, input().split()))
k = int(input())
result = shakutori(input_array, k)
print(f"Number of subarrays with sum <= {k}: {result}")
